csi/node_server.go

```diff
@@ -442,6 +443,14 @@ func (ns *NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 			return nil, status.Errorf(codes.InvalidArgument, "unsupported disk encryption format %v", diskFormat)
 		}
 
+		volumeSize, err := util.ConvertSize(volume.Size)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
+		if volumeSize <= crypto.Luks2MinimalVolumeSize {
+			return nil, status.Errorf(codes.InvalidArgument, "unsupported volume size %v, need to be bigger than 16MiB default LUKS2 header size for encryption", volume.Size)
+		}
+
```

Comment:
Can we move the check to volume validator?

Reply:
Yes, moved it to the volume validator but it would only show the error message one of `longhorn-csi-plugin` pods.